dsl/src/main/kotlin/com/chutneytesting/kotlin/launcher/ExecutionUtils.kt

```diff
@@ -0,0 +1,136 @@
+        scenario: ChutneyScenario,
+        environment: ChutneyEnvironment
+    ): ExecutionRequestDto.StepDefinitionRequestDto {
+
+        val steps = (scenario.givens + scenario.`when` + scenario.thens).filterNotNull()
+        return ExecutionRequestDto.StepDefinitionRequestDto(
+            scenario.title,
+            null,
+            null,
+            "",
+            mapOf(), //inputs
+            mapStepDefinition(steps, environment), // steps
+            mapOf(), //outputs
+            mapOf(), // validations
+            environment.name
+        )
+    }
+
+    private fun mapStepDefinition(
+        steps: List<ChutneyStep>,
+        environment: ChutneyEnvironment
+    ): List<ExecutionRequestDto.StepDefinitionRequestDto> {
+        return steps.map { step ->
+            ExecutionRequestDto.StepDefinitionRequestDto(
+                step.description,
+                mapTargetToTargetExecutionDto(environment, step.implementation?.target),
+                mapStrategyToStrategyExecutionDto(step.strategy),
+                step.implementation?.type,
+                step.implementation?.inputs,
+                mapStepDefinition(step.subSteps, environment), // steps
+                step.implementation?.outputs,  //outputs
+                step.implementation?.validations,  // validations
+                environment.name
+            )
+        }
+    }
+
+    private fun mapStrategyToStrategyExecutionDto(strategy: Strategy?): ExecutionRequestDto.StepStrategyDefinitionRequestDto {
+        return ExecutionRequestDto.StepStrategyDefinitionRequestDto(
+            strategy?.type ?: "",
+            strategy?.parameters ?: emptyMap()
+        )
+    }
+
+    private fun mapTargetToTargetExecutionDto(environment: ChutneyEnvironment, target: String?): TargetExecutionDto? {
+        return environment.findTarget(target)?.let {
+            toTargetExecutionDto(it)
+        }
+    }
+
+    private fun toTargetExecutionDto(target: ChutneyTarget): TargetExecutionDto {
+        return TargetExecutionDto(
+            target.name,
+            target.url,
+            target.configuration.properties,
+            toSecurityDto(target.configuration.security),
+            listOf()
+        )
+    }
+
+    private fun toSecurityDto(securityProperties: ChutneySecurityProperties): SecurityInfoExecutionDto {
+        return SecurityInfoExecutionDto(
+            securityProperties.credential?.username.let {
+                CredentialExecutionDto(
+                    it,
+                    securityProperties.credential?.password
+                )
+            },
+            securityProperties.trustStore?.ifEmpty { null },
+            securityProperties.trustStorePassword?.ifEmpty { null },
+            securityProperties.keyStore?.ifEmpty { null },
+            securityProperties.keyStorePassword?.ifEmpty { null },
+            securityProperties.privateKey?.ifEmpty { null }
+        )
+    }
+}
+
+object ReportWriter {
```

Comment:
ReportWriter is used by the Launcher and the JUnit engine.
Being buried here in launcher package in an "util" class doesn't feel right